server/test/unit/paTest/handleMessageClientsTest.js

```diff
@@ -0,0 +1,268 @@
+// Third-party dependencies
+const { expect, use } = require('chai')
+const { describe, it } = require('mocha')
+const sinon = require('sinon')
+const sinonChai = require('sinon-chai')
+
+// In-house dependencies
+const { helpers, twilioHelpers } = require('brave-alert-lib')
+const { mockResponse } = require('../../testingHelpers')
+const pa = require('../../../pa')
+const db = require('../../../db/db')
+
+const successfulPhoneNumbers = [
+  '+16040000000',
+  '+16041111111',
+  '+16042222222',
+  '+16043333333',
+  '+16044444444',
+  '+16045555555',
+  '+16046666666',
+  '+16047777777',
+  '+16048888888',
+  '+16049999999',
+]
+
+const failPhoneNumbers = [
+  '+16040001111', // for use with testBadFromPhoneNumber
+]
+
+const testFromPhoneNumber = '+17780000000'
+const testBadFromPhoneNumber = '+17781111111'
+const testTwilioMessage = 'Hello, Brave!'
+
+// fake Google ID token; just needs to not be undefined for the validator
+const testGoogleIdToken = 'google-id-token'
+
+// handleMessageClients needs:
+// - id
+// - displayName
+// - responderPhoneNumbers
+// - fallbackPhoneNumbers
+// - heartbeatPhoneNumbers
+// - fromPhoneNumber
+// from each client
+const activeClients = [
+  {
```

Comment:
To better emulate what is returned from `db.getActive Clients` you coule use `factories.clientFactory` in `brave-alert-lib` to make `client` objects, if you want.

Reply:
That is a good point, but my main worry here is that it'll make the file unnecessarily long. I'd probably iterate over the existing activeClients array using 'forEach', then create a new array with the outputs from factories.clientFactory. But this would make no difference, as the only values used by handleMessageClients are those explicitly set in activeClients, so I feel like it's unnecessary overhead. What do you think?

Reply:
Mmm, maybe I'm missing something. It does add more overhead in terms of what the code is doing and the space it takes up in memory, but I don't think it significantly adds to the length of the file. In this array that you're already making, where you have something like 
```
const activeClients = [
  {
    id: 'client-just-responder-phone-number',
    displayName: 'Client Just Responder Phone Number',
    responderPhoneNumbers: [successfulPhoneNumbers[0]],
    fallbackPhoneNumbers: [],
    heartbeatPhoneNumbers: [],
    fromPhoneNumber: testFromPhoneNumber,
  },
  ...
]
```

instead you'd have

```
const activeClients = [
   factories.clientFactory({
    id: 'client-just-responder-phone-number',
    displayName: 'Client Just Responder Phone Number',
    responderPhoneNumbers: [successfulPhoneNumbers[0]],
    fallbackPhoneNumbers: [],
    heartbeatPhoneNumbers: [],
    fromPhoneNumber: testFromPhoneNumber,
  }),
  ...
]
```

One advantage it would give you is that if the client object changes for some reason in a way that is relevant to the code, then you may get those changes for free by using the factory. It will also make it easier to identify where all the client objects are being used if you do need to go through and make manual changes to all of them.

Reply:
Ah, that's a good point, that isn't so bad

Reply:
Done!